File: packages/ruby/lib/logseal/resources/organizations.rb
# frozen_string_literal: true

module LogSeal
  module Resources
    class Organizations
      def initialize(client)
        @client = client
      end

      # List all organizations.
      def list
        @client.request(:get, "/v1/organizations")
      end

      # Create a new organization.
      def create(external_id:, name: nil)
        body = { external_id: }
        body[:name] = name if name
        @client.request(:post, "/v1/organizations", body:)
      end

      # Retrieve an organization by ID.
      def get(id)
        @client.request(:get, "/v1/organizations/#{id}")
      end
    end
  end
end

File: packages/php/src/Resource/Organizations.php
<?php

declare(strict_types=1);

namespace LogSeal\Resource;

use LogSeal\LogSeal;

class Organizations
{
    public function __construct(private readonly LogSeal $client) {}

    public function list(): array
    {
        return $this->client->request('GET', '/v1/organizations');
    }

    public function create(string $externalId, ?string $name = null): array
    {
        $body = ['external_id' => $externalId];
        if ($name !== null) $body['name'] = $name;
        return $this->client->request('POST', '/v1/organizations', $body);
    }

    public function get(string $id): array
    {
        return $this->client->request('GET', "/v1/organizations/{$id}");
    }
}
